src/Common/MoistThermodynamics/MoistThermodynamics.jl

```diff
@@ -779,32 +779,84 @@ Compute the temperature that is consistent with
 See also [`saturation_adjustment`](@ref).
 """
 function saturation_adjustment_q_tot_θ_liq_ice(θ_liq_ice::FT, q_tot::FT, ρ::FT, p::FT) where {FT<:Real}
```

Comment:
Shouldn't you have either pressure `p` or density `rho` as input arguments, but not both, so as not to overspecify the problem?

Reply:
Yes, still in progress..